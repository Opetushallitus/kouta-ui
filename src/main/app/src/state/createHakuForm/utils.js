import get from 'lodash/get';
import pick from 'lodash/pick';

import {
  getKoutaDateString,
  isNumeric,
  formatKoutaDateString,
} from '../../utils';

const DATE_FORMAT = 'DD.MM.YYYY HH:mm';

const getDateTimeValues = dateString => {
  if (!dateString) {
    return { date: '', time: '' };
  }

  const formattedDate = formatKoutaDateString(dateString, DATE_FORMAT);

  const [d = '', t = ''] = formattedDate.split(' ');

  return { date: d, time: t };
};

const getKoutaDateStringByDateTime = ({ date = '', time = '' }) => {
  const [day, month, year] = date.split('.');

  if (!isNumeric(day) || !isNumeric(month) || !isNumeric(year)) {
    return null;
  }

  const [hour, minute] = time.split(':');

  return getKoutaDateString({
    day,
    month,
    year,
    hour: hour || 0,
    minute: minute || 0,
  });
};

export const getHakuByValues = values => {
  const alkamiskausiKoodiUri = get(values, 'aikataulut.kausi') || null;
  const alkamisvuosi = parseInt(get(values, 'aikataulut.vuosi'));
  const kielivalinta = get(values, 'kieliversiot.languages') || [];

  const hakutapaKoodiUri = get(values, 'hakutapa.tapa') || null;

  const hakuajat = (get(values, 'aikataulut.hakuaika') || []).map(
    ({ fromDate, fromTime, toDate, toTime }) => ({
      alkaa: getKoutaDateStringByDateTime({
        date: fromDate,
        time: fromTime,
      }),
      paattyy: getKoutaDateStringByDateTime({
        date: toDate,
        time: toTime,
      }),
    }),
  );

  const hakulomaketyyppi = get(values, 'hakulomake.lomaketyyppi') || null;

  const hakulomake = get(values, 'hakulomake.lomake') || null;

  const hakukohteenLiittamisenTakaraja =
    get(values, 'aikataulut.liittäminen_pvm') &&
    get(values, 'aikataulut.liittäminen_aika')
      ? getKoutaDateStringByDateTime({
          date: values.aikataulut.liittäminen_pvm,
          time: values.aikataulut.liittäminen_aika,
        })
      : null;

  const ajastettuJulkaisu =
    get(values, 'aikataulut.julkaisu_pvm') &&
    get(values, 'aikataulut.julkaisu_aika')
      ? getKoutaDateStringByDateTime({
          date: values.aikataulut.julkaisu_pvm,
          time: values.aikataulut.julkaisu_aika,
        })
      : null;

  const nimi = pick(get(values, 'nimi.nimi') || null, kielivalinta);

  const kohdejoukkoKoodiUri = get(values, 'kohdejoukko.kohde') || null;

  const kohdejoukonTarkenneKoodiUri = null;

  const metadata = {
    tulevaisuudenAikataulu: (get(values, 'aikataulut.aikataulu') || []).map(
      ({ fromDate, fromTime, toDate, toTime }) => ({
        alkaa: getKoutaDateStringByDateTime({
          date: fromDate,
          time: fromTime,
        }),
        paattyy: getKoutaDateStringByDateTime({
          date: toDate,
          time: toTime,
        }),
      }),
    ),
    yhteystieto: {
      nimi: pick(get(values, 'yhteystiedot.nimi') || null, kielivalinta),
      titteli: pick(get(values, 'yhteystiedot.titteli') || null, kielivalinta),
      sahkoposti: pick(get(values, 'yhteystiedot.email') || null, kielivalinta),
      puhelinnumero: pick(
        get(values, 'yhteystiedot.puhelin') || null,
        kielivalinta,
      ),
    },
  };

  const hakukohteenMuokkaamisenTakaraja =
    get(values, 'aikataulut.muokkaus_pvm') &&
    get(values, 'aikataulut.muokkaus_aika')
      ? getKoutaDateStringByDateTime({
          date: values.aikataulut.muokkaus_pvm,
          time: values.aikataulut.muokkaus_aika,
        })
      : null;

  return {
    alkamiskausiKoodiUri,
    kielivalinta,
    hakutapaKoodiUri,
    hakuajat,
    hakukohteenLiittamisenTakaraja,
    nimi,
    kohdejoukkoKoodiUri,
    kohdejoukonTarkenneKoodiUri,
    hakulomaketyyppi,
    metadata,
    hakukohteenMuokkaamisenTakaraja,
    ajastettuJulkaisu,
    alkamisvuosi,
    hakulomake,
  };
};

export const getValuesByHaku = haku => {
  const {
    alkamiskausiKoodiUri = '',
    alkamisvuosi = '',
    hakuajat = [],
    hakutapaKoodiUri = '',
    kohdejoukkoKoodiUri = '',
    hakulomaketyyppi = '',
    hakukohteenLiittamisenTakaraja,
    hakukohteenMuokkaamisenTakaraja,
    ajastettuJulkaisu,
    kielivalinta = [],
    nimi = {},
    metadata = {},
  } = haku;

  const { yhteystieto = {} } = metadata;

  const { tulevaisuudenAikataulu = [] } = metadata;

  const { date: liittäminen_pvm, time: liittäminen_aika } = getDateTimeValues(
    hakukohteenLiittamisenTakaraja,
  );

  const { date: muokkaus_pvm, time: muokkaus_aika } = getDateTimeValues(
    hakukohteenMuokkaamisenTakaraja,
  );

  const { date: julkaisu_pvm, time: julkaisu_aika } = getDateTimeValues(
    ajastettuJulkaisu,
  );

  return {
    nimi: {
      nimi: nimi,
    },
    kieliversiot: {
      languages: kielivalinta,
    },
    aikataulut: {
      kausi: alkamiskausiKoodiUri,
      vuosi: alkamisvuosi ? alkamisvuosi.toString() : '',
      hakuaika: (hakuajat || []).map(({ alkaa, paattyy }) => {
        const { date: fromDate, time: fromTime } = getDateTimeValues(alkaa);
        const { date: toDate, time: toTime } = getDateTimeValues(paattyy);

        return {
          fromDate,
          fromTime,
          toDate,
          toTime,
        };
      }),
      aikataulu: (tulevaisuudenAikataulu || []).map(({ alkaa, paattyy }) => {
        const { date: fromDate, time: fromTime } = getDateTimeValues(alkaa);
        const { date: toDate, time: toTime } = getDateTimeValues(paattyy);

        return {
          fromDate,
          fromTime,
          toDate,
          toTime,
        };
      }),
      liittäminen_pvm: liittäminen_pvm,
      liittäminen_aika: liittäminen_aika,
      muokkaus_pvm: muokkaus_pvm,
      muokkaus_aika: muokkaus_aika,
      julkaisu_pvm: julkaisu_pvm,
      julkaisu_aika: julkaisu_aika,
    },
    hakutapa: {
      tapa: hakutapaKoodiUri,
    },
    kohdejoukko: {
      kohde: kohdejoukkoKoodiUri,
    },
    hakulomake: {
      lomaketyyppi: hakulomaketyyppi,
    },
    yhteystiedot: {
      nimi: get(yhteystieto, 'nimi') || {},
      titteli: get(yhteystieto, 'titteli') || {},
      email: get(yhteystieto, 'sahkoposti') || {},
      puhelin: get(yhteystieto, 'puhelinnumero') || {},
    },
  };
};
