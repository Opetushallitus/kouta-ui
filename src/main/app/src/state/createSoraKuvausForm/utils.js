import get from 'lodash/get';
import pick from 'lodash/pick';
import mapValues from 'lodash/mapValues';

import { JULKAISUTILA } from '../../constants';
import createErrorBuilder from '../../utils/createErrorBuilder';
import parseEditorState from '../../utils/draft/parseEditorState';
import serializeEditorState from '../../utils/draft/serializeEditorState';

const getKielivalinta = values => get(values, 'kieliversiot') || [];

export const getSoraKuvausByValues = values => {
  const kielivalinta = getKielivalinta(values);
  const nimi = pick(get(values, 'tiedot.nimi') || {}, kielivalinta);
  const kuvaus = pick(get(values, 'tiedot.kuvaus') || {}, kielivalinta);
  const koulutustyyppi = get(values, 'koulutustyyppi') || null;
  const julkinen = Boolean(get(values, 'julkisuus'));

  return {
    nimi,
    julkinen,
    koulutustyyppi,
    kielivalinta,
    metadata: {
      kuvaus: mapValues(kuvaus, serializeEditorState),
    },
  };
};

export const getValuesBySoraKuvaus = soraKuvaus => {
  const {
    nimi,
    julkinen,
    koulutustyyppi,
    kielivalinta,
    metadata = {},
  } = soraKuvaus;

  const { kuvaus } = metadata;

  return {
    kieliversiot: kielivalinta || [],
    tiedot: {
      nimi: nimi || {},
      kuvaus: mapValues(kuvaus || {}, parseEditorState),
    },
    julkinen: Boolean(julkinen),
    koulutustyyppi: koulutustyyppi || null,
  };
};

const validateEssentials = ({ values, errorBuilder }) => {
  const kielivalinta = getKielivalinta(values);

  return errorBuilder
    .validateArrayMinLength('kieliversiot', 1)
    .validateTranslations('tiedot.nimi', kielivalinta);
};

const validateCommon = ({ values, errorBuilder }) => {
  const kielivalinta = getKielivalinta(values);

  return errorBuilder
    .validateTranslations('tiedot.kuvaus', kielivalinta)
    .validateExistence('koulutustyyppi');
};

export const validate = ({ tila, values }) => {
  let errorBuilder = createErrorBuilder(values);

  errorBuilder = validateEssentials({ values, errorBuilder });

  if (tila === JULKAISUTILA.TALLENNETTU) {
    return errorBuilder.getErrors();
  }

  errorBuilder = validateCommon({ values, errorBuilder });

  return errorBuilder.getErrors();
};
