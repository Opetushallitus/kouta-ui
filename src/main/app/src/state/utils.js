import { getFormValues, startSubmit, stopSubmit } from 'redux-form';

import { isNonEmptyObject } from '../utils';
import { createSavingErrorToast, createSavingSuccessToast } from './toaster';

const identity = arg => arg;

export const submitAction = ({
  formName,
  validate,
  createSaveAction,
  getValidationArguments = identity,
}) => async (dispatch, getState) => {
  const state = getState();
  const values = getFormValues(formName)(state);
  const errors = validate(getValidationArguments({ values }));

  dispatch(startSubmit(formName));

  if (!isNonEmptyObject(errors)) {
    dispatch(stopSubmit(formName, errors));
    dispatch(createSavingErrorToast());
    return;
  }

  const saveAction = await createSaveAction({ values });

  try {
    const data = await dispatch(saveAction);
    dispatch(stopSubmit(formName));
    dispatch(createSavingSuccessToast());

    return data;
  } catch (e) {
    dispatch(stopSubmit(formName));
    dispatch(createSavingErrorToast());

    throw e;
  }
};
