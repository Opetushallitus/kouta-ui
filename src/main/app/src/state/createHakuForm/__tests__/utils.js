import merge from 'lodash/merge';

import { getHakuByValues, getValuesByHaku } from '../utils';

import { HAKULOMAKETYYPPI } from '../../../constants';

const baseValues = {
  nimi: {
    fi: 'Nimi',
    sv: 'Namn',
  },
  kieliversiot: ['fi', 'sv'],
  aikataulut: {
    kausi: 'alkamiskausi_1#1',
    vuosi: { value: '2020' },
    hakuaika: [
      {
        alkaa: '2019-04-16T08:44',
        paattyy: '2019-04-18T08:44',
      },
      {
        alkaa: '2019-06-16T08:44',
        paattyy: '2019-06-18T08:44',
      },
    ],
    aikataulu: [
      {
        alkaa: '2019-08-16T08:44',
        paattyy: '2019-08-18T08:44',
      },
      {
        alkaa: '2019-09-16T08:44',
        paattyy: '2019-09-18T08:44',
      },
    ],
    lisaamisenTakaraja: '2019-09-16T08:44',
    muokkauksenTakaraja: '2019-10-16T08:44',
    ajastettuJulkaisu: '2019-11-16T08:44',
  },
  hakutapa: 'hakutapa_1#1',
  kohdejoukko: 'kohde_1#1',
  hakulomake: {
    tyyppi: HAKULOMAKETYYPPI.ATARU,
    lomake: {
      [HAKULOMAKETYYPPI.ATARU]: { value: '12345' },
    },
  },
  yhteyshenkilot: [
    {
      nimi: {
        fi: 'Fi nimi',
        sv: 'Sv nimi',
      },
      titteli: { fi: 'Fi titteli', sv: 'Sv titteli' },
      sahkoposti: { fi: 'Fi sähköposti', sv: 'Sv sähköposti' },
      puhelinnumero: { fi: 'Fi puhelinnumero', sv: 'Sv puhelinnumero' },
      verkkosivu: { fi: 'Fi verkkosivu', sv: 'Sv verkkosivu' },
    },
  ],
  valintakoe: {
    tyypit: [{ value: 'tyyppi_1#1' }],
    tilaisuudet: {
      'tyyppi_1#1': [
        {
          osoite: { fi: 'fi osoite', sv: 'sv osoite' },
          postinumero: '00520',
          postitoimipaikka: {
            fi: 'fi posititoimipaikka',
            sv: 'sv posititoimipaikka',
          },
          alkaa: '2019-04-16T08:44',
          paattyy: '2019-04-18T08:44',
          lisatietoja: {
            fi: 'fi lisatietoja',
            sv: 'sv lisatietoja',
          },
        },
      ],
    },
  },
};

const baseHaku = {
  alkamiskausiKoodiUri: 'kausi_1#1',
  kielivalinta: ['fi', 'sv'],
  hakutapaKoodiUri: 'hakutapa_1#1',
  hakuajat: [{ alkaa: '2019-03-29T12:28', paattyy: '2019-09-29T12:30' }],
  hakukohteenLiittamisenTakaraja: '2019-03-29T12:28',
  nimi: {
    fi: 'Fi nimi',
    sv: 'Sv nimi',
  },
  kohdejoukkoKoodiUri: 'kohdejoukko_1#1',
  kohdejoukonTarkenneKoodiUri: 'tarkenne_1#1',
  metadata: {
    tulevaisuudenAikataulu: [
      { alkaa: '2019-02-20T12:28', paattyy: '2019-08-01T12:45' },
    ],
    yhteyshenkilot: [
      {
        nimi: {
          fi: 'Fi nimi',
          sv: 'Sv nimi',
        },
        titteli: { fi: 'Fi titteli', sv: 'Sv titteli' },
        sahkoposti: { fi: 'Fi sähköposti', sv: 'Sv sähköposti' },
        puhelinnumero: { fi: 'Fi puhelinnumero', sv: 'Sv puhelinnumero' },
        wwwSivu: { fi: 'Fi verkkosivu', sv: 'Sv verkkosivu' },
      },
    ],
  },
  hakukohteenMuokkaamisenTakaraja: '2019-03-11T09:45',
  ajastettuJulkaisu: '2020-01-20T05:00',
  alkamisvuosi: 2020,
  hakulomaketyyppi: HAKULOMAKETYYPPI.ATARU,
  hakulomakeId: '12345',
  valintakokeet: [
    {
      tyyppi: 'tyyppi_1#1',
      tilaisuudet: [
        {
          osoite: {
            osoite: { fi: 'fi osoite', sv: 'sv osoite' },
            postinumero: '00520',
            postitoimipaikka: {
              fi: 'fi posititoimipaikka',
              sv: 'sv posititoimipaikka',
            },
          },
          aika: {
            alkaa: '2019-04-16T08:44',
            paattyy: '2019-04-18T08:44',
          },
          lisatietoja: {
            fi: 'fi lisatietoja',
            sv: 'sv lisatietoja',
          },
        },
      ],
    },
  ],
};

test('getHakuByValues returns correct haku given form values', () => {
  const haku = getHakuByValues(baseValues);

  expect(haku).toMatchSnapshot();
});

test('getHakuByValues returns correct haku given different hakulomake variations', () => {
  const hakuMuu = getHakuByValues(
    merge({}, baseValues, {
      hakulomake: {
        tyyppi: HAKULOMAKETYYPPI.MUU,
        linkki: { fi: 'https://google.fi' },
      },
    }),
  );

  const hakuEiHakua = getHakuByValues(
    merge({}, baseValues, {
      hakulomake: {
        tyyppi: HAKULOMAKETYYPPI.EI_SAHKOISTA_HAKUA,
        kuvaus: { fi: 'kuvaus' },
      },
    }),
  );

  expect(hakuMuu).toMatchSnapshot();
  expect(hakuEiHakua).toMatchSnapshot();
});

test('getValuesByHaku returns correct form values given haku', () => {
  const values = getValuesByHaku(baseHaku);

  expect(values).toMatchSnapshot();
});

test('getValuesByHaku returns correct form values given different hakulomake variations', () => {
  const valuesMuu = getValuesByHaku(
    merge({}, baseHaku, {
      hakulomaketyyppi: HAKULOMAKETYYPPI.MUU,
      hakulomakeLinkki: {
        fi: 'https://google.fi',
      },
    }),
  );

  const valuesEiHakua = getValuesByHaku(
    merge({}, baseHaku, {
      hakulomaketyyppi: HAKULOMAKETYYPPI.EI_SAHKOISTA_HAKUA,
      hakulomakeKuvaus: {
        fi: 'kuvaus',
      },
    }),
  );

  expect(valuesMuu).toMatchSnapshot();
  expect(valuesEiHakua).toMatchSnapshot();
});
