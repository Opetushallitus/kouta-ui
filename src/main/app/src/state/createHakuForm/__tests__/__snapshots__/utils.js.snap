// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getHakuByValues returns correct haku given different hakulomake variations 1`] = `
Object {
  "ajastettuJulkaisu": "2019-11-16T08:44",
  "alkamiskausiKoodiUri": "alkamiskausi_1#1",
  "alkamisvuosi": 2020,
  "hakuajat": Array [
    Object {
      "alkaa": "2019-04-16T08:44",
      "paattyy": "2019-04-18T08:44",
    },
    Object {
      "alkaa": "2019-06-16T08:44",
      "paattyy": "2019-06-18T08:44",
    },
  ],
  "hakukohteenLiittamisenTakaraja": "2019-09-16T08:44",
  "hakukohteenMuokkaamisenTakaraja": "2019-10-16T08:44",
  "hakulomakeId": null,
  "hakulomakeKuvaus": Object {},
  "hakulomakeLinkki": Object {
    "fi": "https://google.fi",
  },
  "hakulomaketyyppi": "muu",
  "hakutapaKoodiUri": "hakutapa_1#1",
  "kielivalinta": Array [
    "fi",
    "sv",
  ],
  "kohdejoukkoKoodiUri": "kohde_1#1",
  "kohdejoukonTarkenneKoodiUri": null,
  "metadata": Object {
    "tulevaisuudenAikataulu": Array [
      Object {
        "alkaa": "2019-08-16T08:44",
        "paattyy": "2019-08-18T08:44",
      },
      Object {
        "alkaa": "2019-09-16T08:44",
        "paattyy": "2019-09-18T08:44",
      },
    ],
    "yhteyshenkilot": Array [
      Object {
        "nimi": Object {
          "fi": "Fi nimi",
          "sv": "Sv nimi",
        },
        "puhelinnumero": Object {
          "fi": "Fi puhelinnumero",
          "sv": "Sv puhelinnumero",
        },
        "sahkoposti": Object {
          "fi": "Fi sähköposti",
          "sv": "Sv sähköposti",
        },
        "titteli": Object {
          "fi": "Fi titteli",
          "sv": "Sv titteli",
        },
        "wwwSivu": Object {
          "fi": "Fi verkkosivu",
          "sv": "Sv verkkosivu",
        },
      },
    ],
  },
  "nimi": Object {
    "fi": "Nimi",
    "sv": "Namn",
  },
  "valintakokeet": Array [
    Object {
      "tilaisuudet": Array [
        Object {
          "aika": Object {
            "alkaa": "2019-04-16T08:44",
            "paattyy": "2019-04-18T08:44",
          },
          "lisatietoja": Object {
            "fi": "fi lisatietoja",
            "sv": "sv lisatietoja",
          },
          "osoite": Object {
            "osoite": Object {
              "fi": "fi osoite",
              "sv": "sv osoite",
            },
            "postinumero": "00520",
            "postitoimipaikka": Object {
              "fi": "fi posititoimipaikka",
              "sv": "sv posititoimipaikka",
            },
          },
        },
      ],
      "tyyppi": "tyyppi_1#1",
    },
  ],
}
`;

exports[`getHakuByValues returns correct haku given different hakulomake variations 2`] = `
Object {
  "ajastettuJulkaisu": "2019-11-16T08:44",
  "alkamiskausiKoodiUri": "alkamiskausi_1#1",
  "alkamisvuosi": 2020,
  "hakuajat": Array [
    Object {
      "alkaa": "2019-04-16T08:44",
      "paattyy": "2019-04-18T08:44",
    },
    Object {
      "alkaa": "2019-06-16T08:44",
      "paattyy": "2019-06-18T08:44",
    },
  ],
  "hakukohteenLiittamisenTakaraja": "2019-09-16T08:44",
  "hakukohteenMuokkaamisenTakaraja": "2019-10-16T08:44",
  "hakulomakeId": null,
  "hakulomakeKuvaus": Object {
    "fi": "kuvaus",
  },
  "hakulomakeLinkki": Object {},
  "hakulomaketyyppi": "ei sähköistä",
  "hakutapaKoodiUri": "hakutapa_1#1",
  "kielivalinta": Array [
    "fi",
    "sv",
  ],
  "kohdejoukkoKoodiUri": "kohde_1#1",
  "kohdejoukonTarkenneKoodiUri": null,
  "metadata": Object {
    "tulevaisuudenAikataulu": Array [
      Object {
        "alkaa": "2019-08-16T08:44",
        "paattyy": "2019-08-18T08:44",
      },
      Object {
        "alkaa": "2019-09-16T08:44",
        "paattyy": "2019-09-18T08:44",
      },
    ],
    "yhteyshenkilot": Array [
      Object {
        "nimi": Object {
          "fi": "Fi nimi",
          "sv": "Sv nimi",
        },
        "puhelinnumero": Object {
          "fi": "Fi puhelinnumero",
          "sv": "Sv puhelinnumero",
        },
        "sahkoposti": Object {
          "fi": "Fi sähköposti",
          "sv": "Sv sähköposti",
        },
        "titteli": Object {
          "fi": "Fi titteli",
          "sv": "Sv titteli",
        },
        "wwwSivu": Object {
          "fi": "Fi verkkosivu",
          "sv": "Sv verkkosivu",
        },
      },
    ],
  },
  "nimi": Object {
    "fi": "Nimi",
    "sv": "Namn",
  },
  "valintakokeet": Array [
    Object {
      "tilaisuudet": Array [
        Object {
          "aika": Object {
            "alkaa": "2019-04-16T08:44",
            "paattyy": "2019-04-18T08:44",
          },
          "lisatietoja": Object {
            "fi": "fi lisatietoja",
            "sv": "sv lisatietoja",
          },
          "osoite": Object {
            "osoite": Object {
              "fi": "fi osoite",
              "sv": "sv osoite",
            },
            "postinumero": "00520",
            "postitoimipaikka": Object {
              "fi": "fi posititoimipaikka",
              "sv": "sv posititoimipaikka",
            },
          },
        },
      ],
      "tyyppi": "tyyppi_1#1",
    },
  ],
}
`;

exports[`getHakuByValues returns correct haku given form values 1`] = `
Object {
  "ajastettuJulkaisu": "2019-11-16T08:44",
  "alkamiskausiKoodiUri": "alkamiskausi_1#1",
  "alkamisvuosi": 2020,
  "hakuajat": Array [
    Object {
      "alkaa": "2019-04-16T08:44",
      "paattyy": "2019-04-18T08:44",
    },
    Object {
      "alkaa": "2019-06-16T08:44",
      "paattyy": "2019-06-18T08:44",
    },
  ],
  "hakukohteenLiittamisenTakaraja": "2019-09-16T08:44",
  "hakukohteenMuokkaamisenTakaraja": "2019-10-16T08:44",
  "hakulomakeId": "12345",
  "hakulomakeKuvaus": Object {},
  "hakulomakeLinkki": Object {},
  "hakulomaketyyppi": "ataru",
  "hakutapaKoodiUri": "hakutapa_1#1",
  "kielivalinta": Array [
    "fi",
    "sv",
  ],
  "kohdejoukkoKoodiUri": "kohde_1#1",
  "kohdejoukonTarkenneKoodiUri": null,
  "metadata": Object {
    "tulevaisuudenAikataulu": Array [
      Object {
        "alkaa": "2019-08-16T08:44",
        "paattyy": "2019-08-18T08:44",
      },
      Object {
        "alkaa": "2019-09-16T08:44",
        "paattyy": "2019-09-18T08:44",
      },
    ],
    "yhteyshenkilot": Array [
      Object {
        "nimi": Object {
          "fi": "Fi nimi",
          "sv": "Sv nimi",
        },
        "puhelinnumero": Object {
          "fi": "Fi puhelinnumero",
          "sv": "Sv puhelinnumero",
        },
        "sahkoposti": Object {
          "fi": "Fi sähköposti",
          "sv": "Sv sähköposti",
        },
        "titteli": Object {
          "fi": "Fi titteli",
          "sv": "Sv titteli",
        },
        "wwwSivu": Object {
          "fi": "Fi verkkosivu",
          "sv": "Sv verkkosivu",
        },
      },
    ],
  },
  "nimi": Object {
    "fi": "Nimi",
    "sv": "Namn",
  },
  "valintakokeet": Array [
    Object {
      "tilaisuudet": Array [
        Object {
          "aika": Object {
            "alkaa": "2019-04-16T08:44",
            "paattyy": "2019-04-18T08:44",
          },
          "lisatietoja": Object {
            "fi": "fi lisatietoja",
            "sv": "sv lisatietoja",
          },
          "osoite": Object {
            "osoite": Object {
              "fi": "fi osoite",
              "sv": "sv osoite",
            },
            "postinumero": "00520",
            "postitoimipaikka": Object {
              "fi": "fi posititoimipaikka",
              "sv": "sv posititoimipaikka",
            },
          },
        },
      ],
      "tyyppi": "tyyppi_1#1",
    },
  ],
}
`;

exports[`getValuesByHaku returns correct form values given different hakulomake variations 1`] = `
Object {
  "aikataulut": Object {
    "aikataulu": Array [
      Object {
        "alkaa": "2019-02-20T12:28",
        "paattyy": "2019-08-01T12:45",
      },
    ],
    "ajastettuJulkaisu": "2020-01-20T05:00",
    "hakuaika": Array [
      Object {
        "alkaa": "2019-03-29T12:28",
        "paattyy": "2019-09-29T12:30",
      },
    ],
    "kausi": "kausi_1#1",
    "lisaamisenTakaraja": "2019-03-29T12:28",
    "muokkauksenTakaraja": "2019-03-11T09:45",
    "vuosi": Object {
      "value": "2020",
    },
  },
  "hakulomake": Object {
    "kuvaus": Object {},
    "linkki": Object {
      "fi": "https://google.fi",
    },
    "lomake": Object {},
    "tyyppi": "muu",
  },
  "hakutapa": "hakutapa_1#1",
  "kieliversiot": Array [
    "fi",
    "sv",
  ],
  "kohdejoukko": "kohdejoukko_1#1",
  "nimi": Object {
    "fi": "Fi nimi",
    "sv": "Sv nimi",
  },
  "valintakoe": Object {
    "tilaisuudet": Object {
      "tyyppi_1#1": Array [
        Object {
          "alkaa": "2019-04-16T08:44",
          "lisatietoja": Object {
            "fi": "fi lisatietoja",
            "sv": "sv lisatietoja",
          },
          "osoite": Object {
            "fi": "fi osoite",
            "sv": "sv osoite",
          },
          "paattyy": "2019-04-18T08:44",
          "postinumero": "00520",
          "postitoimipaikka": Object {
            "fi": "fi posititoimipaikka",
            "sv": "sv posititoimipaikka",
          },
        },
      ],
    },
    "tyypit": Array [
      Object {
        "value": "tyyppi_1#1",
      },
    ],
  },
  "yhteyshenkilot": Array [
    Object {
      "nimi": Object {
        "fi": "Fi nimi",
        "sv": "Sv nimi",
      },
      "puhelinnumero": Object {
        "fi": "Fi puhelinnumero",
        "sv": "Sv puhelinnumero",
      },
      "sahkoposti": Object {
        "fi": "Fi sähköposti",
        "sv": "Sv sähköposti",
      },
      "titteli": Object {
        "fi": "Fi titteli",
        "sv": "Sv titteli",
      },
      "verkkosivu": Object {
        "fi": "Fi verkkosivu",
        "sv": "Sv verkkosivu",
      },
    },
  ],
}
`;

exports[`getValuesByHaku returns correct form values given different hakulomake variations 2`] = `
Object {
  "aikataulut": Object {
    "aikataulu": Array [
      Object {
        "alkaa": "2019-02-20T12:28",
        "paattyy": "2019-08-01T12:45",
      },
    ],
    "ajastettuJulkaisu": "2020-01-20T05:00",
    "hakuaika": Array [
      Object {
        "alkaa": "2019-03-29T12:28",
        "paattyy": "2019-09-29T12:30",
      },
    ],
    "kausi": "kausi_1#1",
    "lisaamisenTakaraja": "2019-03-29T12:28",
    "muokkauksenTakaraja": "2019-03-11T09:45",
    "vuosi": Object {
      "value": "2020",
    },
  },
  "hakulomake": Object {
    "kuvaus": Object {
      "fi": "kuvaus",
    },
    "linkki": Object {},
    "lomake": Object {},
    "tyyppi": "ei sähköistä",
  },
  "hakutapa": "hakutapa_1#1",
  "kieliversiot": Array [
    "fi",
    "sv",
  ],
  "kohdejoukko": "kohdejoukko_1#1",
  "nimi": Object {
    "fi": "Fi nimi",
    "sv": "Sv nimi",
  },
  "valintakoe": Object {
    "tilaisuudet": Object {
      "tyyppi_1#1": Array [
        Object {
          "alkaa": "2019-04-16T08:44",
          "lisatietoja": Object {
            "fi": "fi lisatietoja",
            "sv": "sv lisatietoja",
          },
          "osoite": Object {
            "fi": "fi osoite",
            "sv": "sv osoite",
          },
          "paattyy": "2019-04-18T08:44",
          "postinumero": "00520",
          "postitoimipaikka": Object {
            "fi": "fi posititoimipaikka",
            "sv": "sv posititoimipaikka",
          },
        },
      ],
    },
    "tyypit": Array [
      Object {
        "value": "tyyppi_1#1",
      },
    ],
  },
  "yhteyshenkilot": Array [
    Object {
      "nimi": Object {
        "fi": "Fi nimi",
        "sv": "Sv nimi",
      },
      "puhelinnumero": Object {
        "fi": "Fi puhelinnumero",
        "sv": "Sv puhelinnumero",
      },
      "sahkoposti": Object {
        "fi": "Fi sähköposti",
        "sv": "Sv sähköposti",
      },
      "titteli": Object {
        "fi": "Fi titteli",
        "sv": "Sv titteli",
      },
      "verkkosivu": Object {
        "fi": "Fi verkkosivu",
        "sv": "Sv verkkosivu",
      },
    },
  ],
}
`;

exports[`getValuesByHaku returns correct form values given haku 1`] = `
Object {
  "aikataulut": Object {
    "aikataulu": Array [
      Object {
        "alkaa": "2019-02-20T12:28",
        "paattyy": "2019-08-01T12:45",
      },
    ],
    "ajastettuJulkaisu": "2020-01-20T05:00",
    "hakuaika": Array [
      Object {
        "alkaa": "2019-03-29T12:28",
        "paattyy": "2019-09-29T12:30",
      },
    ],
    "kausi": "kausi_1#1",
    "lisaamisenTakaraja": "2019-03-29T12:28",
    "muokkauksenTakaraja": "2019-03-11T09:45",
    "vuosi": Object {
      "value": "2020",
    },
  },
  "hakulomake": Object {
    "kuvaus": Object {},
    "linkki": Object {},
    "lomake": Object {
      "ataru": Object {
        "value": "12345",
      },
    },
    "tyyppi": "ataru",
  },
  "hakutapa": "hakutapa_1#1",
  "kieliversiot": Array [
    "fi",
    "sv",
  ],
  "kohdejoukko": "kohdejoukko_1#1",
  "nimi": Object {
    "fi": "Fi nimi",
    "sv": "Sv nimi",
  },
  "valintakoe": Object {
    "tilaisuudet": Object {
      "tyyppi_1#1": Array [
        Object {
          "alkaa": "2019-04-16T08:44",
          "lisatietoja": Object {
            "fi": "fi lisatietoja",
            "sv": "sv lisatietoja",
          },
          "osoite": Object {
            "fi": "fi osoite",
            "sv": "sv osoite",
          },
          "paattyy": "2019-04-18T08:44",
          "postinumero": "00520",
          "postitoimipaikka": Object {
            "fi": "fi posititoimipaikka",
            "sv": "sv posititoimipaikka",
          },
        },
      ],
    },
    "tyypit": Array [
      Object {
        "value": "tyyppi_1#1",
      },
    ],
  },
  "yhteyshenkilot": Array [
    Object {
      "nimi": Object {
        "fi": "Fi nimi",
        "sv": "Sv nimi",
      },
      "puhelinnumero": Object {
        "fi": "Fi puhelinnumero",
        "sv": "Sv puhelinnumero",
      },
      "sahkoposti": Object {
        "fi": "Fi sähköposti",
        "sv": "Sv sähköposti",
      },
      "titteli": Object {
        "fi": "Fi titteli",
        "sv": "Sv titteli",
      },
      "verkkosivu": Object {
        "fi": "Fi verkkosivu",
        "sv": "Sv verkkosivu",
      },
    },
  ],
}
`;
